core/scheduler/service/replay_service.go

```diff
@@ -0,0 +1,56 @@
+package service
+
+import (
+	"fmt"
+
+	"github.com/google/uuid"
+	"golang.org/x/net/context"
+
+	"github.com/odpf/optimus/core/scheduler"
+	"github.com/odpf/optimus/core/tenant"
+	"github.com/odpf/optimus/internal/lib/cron"
+)
+
+type ReplayRepository interface {
+	RegisterReplay(ctx context.Context, replay *scheduler.Replay, runs []*scheduler.JobRunStatus) (uuid.UUID, error)
+	UpdateReplay(ctx context.Context, replayID uuid.UUID, state scheduler.ReplayState, runs []*scheduler.JobRunStatus, message string) error
+	UpdateReplayStatus(ctx context.Context, replayID uuid.UUID, state scheduler.ReplayState, message string) error
+
+	GetReplayToExecute(context.Context) (*scheduler.ReplayWithRun, error)
+	GetReplayRequestsByStatus(ctx context.Context, statusList []scheduler.ReplayState) ([]*scheduler.Replay, error)
+}
+
+type ReplayValidator interface {
+	Validate(ctx context.Context, replayRequest *scheduler.Replay, jobCron *cron.ScheduleSpec) error
+}
+
+type ReplayService struct {
+	replayRepo ReplayRepository
+	jobRepo    JobRepository
+
+	validator ReplayValidator
+}
+
+func (r ReplayService) CreateReplay(ctx context.Context, tenant tenant.Tenant, jobName scheduler.JobName, config *scheduler.ReplayConfig) (replayID uuid.UUID, err error) {
+	subjectJob, err := r.jobRepo.GetJobDetails(ctx, tenant.ProjectName(), jobName)
+	if err != nil {
+		return uuid.Nil, fmt.Errorf("unable to get job details from DB for jobName: %s, project:%s,  error:%w ", jobName, tenant.ProjectName().String(), err)
+	}
+
+	jobCron, err := cron.ParseCronSchedule(subjectJob.Schedule.Interval)
```

Comment:
in which scenario job is add but cron can't be parsed

Reply:
if this is not expected just add explicitly this shouldn't happen in a realistic scenario

Reply:
in the error message, should mention this is not the expected scenario.